src/GraphQL.NewtonsoftJson/InputsConverter.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace GraphQL.NewtonsoftJson
+{
+    /// <summary>
+    /// A custom JsonConverter for reading an <see cref="Inputs"/> object.
+    /// Doesn't support write.
+    /// </summary>
+    public class InputsConverter : JsonConverter
+    {
+        /// <inheritdoc/>
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+            => ReadDictionary(reader).ToInputs();
+
+        internal static Dictionary<string, object> ReadDictionary(JsonReader reader)
+        {
+            if (reader.TokenType != JsonToken.StartObject)
+                throw new JsonException();
+
+            var result = new Dictionary<string, object>();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonToken.EndObject)
+                    break;
+
+                if (reader.TokenType != JsonToken.PropertyName)
+                    throw new JsonException();
+
+                string key = (string)reader.Value;
+
+                // move to property value
+                if (!reader.Read())
+                    throw new JsonException();
+
+                result.Add(key, ReadValue(reader));
+            }
+
+            return result;
+        }
+
+        private static object ReadValue(JsonReader reader)
+            => reader.TokenType switch
+            {
+                JsonToken.StartArray => ReadArray(reader),
+                JsonToken.StartObject => ReadDictionary(reader),
+                JsonToken.Integer => ReadNumber(reader),
+                JsonToken.Float => reader.Value,
+                JsonToken.Boolean => reader.Value,
+                JsonToken.String => reader.Value,
+                JsonToken.Null => null,
+                JsonToken.None => null,
```

Comment:
```suggestion
```

 //
        // Summary:
        //     This is returned by the Newtonsoft.Json.JsonReader if a read method has not been
        //     called.
        None = 0,